File: seongwon/Programmers/Lv2_귤_고르기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    vector<int> arr(10000001,0);
    for(int i = 0; i < tangerine.size(); i++){
        arr[tangerine[i]]++;
    }
    sort(arr.begin(), arr.end());
    reverse(arr.begin(), arr.end());  //내림차순으로 뒤집음
    int a = 0;
    for(int i : arr){
        a += i;
        answer++;
        if(a >= k){
            break;
        }
    }
    return answer;
}

File: chaerim/Programmers/Level_2/Lv2_귤_고르기.py
def solution(k, tangerine):
    answer = 0

    count_dict = {}
    for size in tangerine:
        if size in count_dict:
            count_dict[size] += 1
        else:
            count_dict[size] = 1

    sorted_counts = sorted(count_dict.values(), reverse=True)

    for count in sorted_counts:
        if k > 0:
            k -= count
            answer += 1
        else:
            break

    return answer